AnnotatoBackend/Sources/App/Models/Extensions/DocumentEntity+CRUD.swift

```diff
@@ -20,12 +20,17 @@ extension DocumentEntity {
     ///   - tx: The database instance in a transaction.
     ///   - document: The updated Document instance.
     func customUpdate(on tx: Database, usingUpdatedModel document: Document) async throws {
```

Comment:
Regarding my first comment after you requested review, I think the `deletedAt` is not updated because we do not copy the properties in `copyPropertiesOf` method.

This applies for all entities